gui/backend/upsert_bot_config.go

```diff
@@ -144,6 +145,15 @@ func (s *APIServer) validateConfigs(req upsertBotConfigRequest) *upsertBotConfig
 		}
 	}
 
+	isValid, e := isBotNameValid(req.Name)
+	if !isValid {
+		if e != nil {
+			errResp.Name = "unable to parse bot name: internal server error"
```

Comment:
this is not an acceptable error to show the user on the input value. There needs to be a better flow.
with the changes to how the regex compile should work (comment below) this is not a situation we have to worry about now thankfully.